ext/crypto/encrypt.rs

```diff
@@ -0,0 +1,138 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use crate::shared::*;
+use block_modes::BlockMode;
+use deno_core::error::type_error;
+use deno_core::error::AnyError;
+use deno_core::OpState;
+use deno_core::ZeroCopyBuf;
+use rand::rngs::OsRng;
+use rsa::pkcs1::FromRsaPublicKey;
+use rsa::PaddingScheme;
+use rsa::PublicKey;
+use serde::Deserialize;
+use sha1::Digest;
+use sha1::Sha1;
+use sha2::Sha256;
+use sha2::Sha384;
+use sha2::Sha512;
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EncryptOptions {
+  key: RawKeyData,
+  #[serde(flatten)]
+  algorithm: EncryptAlgorithm,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase", tag = "algorithm")]
+pub enum EncryptAlgorithm {
+  #[serde(rename = "RSA-OAEP")]
+  RsaOaep {
+    hash: ShaHash,
+    #[serde(with = "serde_bytes")]
+    label: Vec<u8>,
+  },
+  #[serde(rename = "AES", rename_all = "camelCase")]
+  AesCbc {
+    #[serde(with = "serde_bytes")]
+    iv: Vec<u8>,
+    length: usize,
+  },
+}
+pub async fn op_crypto_encrypt(
+  _state: Rc<RefCell<OpState>>,
+  opts: EncryptOptions,
+  data: ZeroCopyBuf,
+) -> Result<ZeroCopyBuf, AnyError> {
+  let key = opts.key;
+  let fun = move || match opts.algorithm {
+    EncryptAlgorithm::RsaOaep { hash, label } => {
+      encrypt_rsa_oaep(key, hash, label, &data)
+    }
+    EncryptAlgorithm::AesCbc { iv, length } => {
+      encrypt_aes_cbc(key, length, iv, &data)
+    }
+  };
+  let buf = tokio::task::spawn_blocking(fun).await.unwrap()?;
+  Ok(buf.into())
+}
+
+fn encrypt_rsa_oaep(
+  key: RawKeyData,
+  hash: ShaHash,
+  label: Vec<u8>,
+  data: &[u8],
+) -> Result<Vec<u8>, deno_core::anyhow::Error> {
+  let label = String::from_utf8_lossy(&label).to_string();
```

Comment:
```suggestion
  let label = String::from_utf8_lossy(&label).into_owned();
```
Otherwise the Cow always heap-allocates, I think?

Reply:
`Cow.into_owned` returns a `Cow` not a `String`. `Cow<str>` turning into `String` would require specialization I think.

Reply:
That doesn't sound right. `Cow::into_owned()` returns `ToOwned::Owned`, which for `str` is `String` (because `str` implements `ToOwned`.)

Reply:
My compiler disagrees. When using `to_owned`:

```
error[E0308]: mismatched types
  --> ext/crypto/encrypt.rs:79:19
   |
79 |       label: Some(label),
   |                   ^^^^^- help: try using a conversion method: `.to_string()`
   |                   |
   |                   expected struct `std::string::String`, found enum `Cow`
   |
   = note: expected struct `std::string::String`
                found enum `Cow<'_, str>`
```

Reply:
/me whispers *in*to_owned :)